### PR TITLE
Updated UCF101 to class format

### PR DESCRIPTION
**Pull request by davidslater**
Open issues:
1) performance and attacks are based on the split versions of the videos.
2) have to reinstantiate ART attacks for each batch due to different sizes (I don't know how ART works with different sizes of inputs)
3) the metrics need to be unified, but I think that we should punt on this until after ImageClassification unification.

### CODE REVIEW

File: armory/scenarios/ucf101.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        video_count = 0
+        for _ in tqdm(range(len(test_data_generator) // 10), desc="Attack"):
+            x_tests, y_tests = test_data_generator.get_batch()
+            for x_test, y_test in zip(x_tests, y_tests):
+                # each x_test is of shape (n_stack, 3, 16, 112, 112)
+                attack = attack_fn(
+                    classifier=classifier,
+                    **attack_config["kwargs"],
+                    batch_size=x_test.shape[0],
+                )
```
Comment (seanpmorgan) [inline armory/scenarios/ucf101.py:164]: `attack.set_params(batch_size=x_test.shape[0])`

Allows for this to be changed in iteration and loaded once using `load_model`

File: armory/scenarios/ucf101.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+            preprocessing_fn=preprocessing_fn,
+        )
+
+        test_accuracies = AverageMeter()
+        test_accuracies_top5 = AverageMeter()
+        video_count = 0
+        for x_tests, y_tests in tqdm(test_data_generator, desc="Benign"):
+            for x_test, y_test in zip(x_tests, y_tests):
+                y = classifier.predict(x_test)
+                y = np.argsort(np.mean(y, axis=0))[-5:][::-1]
```
Comment (seanpmorgan) [inline armory/scenarios/ucf101.py:110]: Unclear of where `[-5:]` is coming from?

File: armory/scenarios/ucf101.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+            config["dataset"],
+            epochs=1,
+            split_type="test",
+            preprocessing_fn=preprocessing_fn,
+        )
+
+        adv_accuracies = AverageMeter()
+        adv_accuracies_top5 = AverageMeter()
+        video_count = 0
+        for _ in tqdm(range(len(test_data_generator) // 10), desc="Attack"):
```
Comment (seanpmorgan) [inline armory/scenarios/ucf101.py:156]: Why `//10`? Just to make it quicker?

File: armory/scenarios/ucf101.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+                    for i, xt in enumerate(x_trains):
+                        rand_stack = np.random.randint(0, xt.shape[0])
+                        x_train[i, ...] = xt[rand_stack, ...]
+                    classifier.fit(
+                        x_train, y_trains, batch_size=batch_size, nb_epochs=1
+                    )
+                logger.info(f"Epoch {epoch}/{train_epochs} took {time.time() - start}s")
+
+                # evaluate on test examples
+                classifier.set_learning_phase(False)
```
[the remark this answers is not in the document]
Comment (davidslater) [reply armory/scenarios/ucf101.py:68]: There are a number of models that can do things differently during prediction. For PyTorch, I think this basically sets `torch.no_grad()` so that it doesn't keep track of gradients while computing.

File: armory/scenarios/ucf101.py [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+                test_data_generator = load_dataset(
+                    config["dataset"],
+                    epochs=1,
+                    split_type="test",
+                    preprocessing_fn=preprocessing_fn,
+                )
+
+                accuracies = AverageMeter()
+                accuracies_top5 = AverageMeter()
+                video_count = 0
```
[the remark this answers is not in the document]
Comment (davidslater) [reply armory/scenarios/ucf101.py:78]: Will remove

File: armory/scenarios/ucf101.py (hunk 2 above)
Comment (davidslater) [reply armory/scenarios/ucf101.py:110]: It's calculating top_5 accuracy - I'd like to put this into the `armory.util.metrics` file and call it, though that may want to wait for the ImageClassification unification.

File: armory/scenarios/ucf101.py (hunk 3 above)
Comment (davidslater) [reply armory/scenarios/ucf101.py:156]: It was from the original. My guess is yes, to make it quicker.

Comment (davidslater) [reply armory/scenarios/ucf101.py:156]: I think that we should have a (configurable) way to subsample without hard coding into the scenario. Thoughts?

File: armory/scenarios/ucf101.py (hunk 1 above)
Comment (davidslater) [reply armory/scenarios/ucf101.py:164]: Great!

**Comment by davidslater** [comment]
@seanpmorgan I think that should handle your concerns. Ready for another review.
